nucypher/crypto/powers.py

```diff
@@ -141,26 +142,29 @@ def account(self) -> str:
 
     def activate(self, password: str = None):
         """Be Consumed"""
-        self.unlock_account(password=password or self.__password)
-        self.__password = None
+        self.unlock_account(password=password)
+        if self.__cache is False:
+            self.__password = None
         self.blockchain.transacting_power = self
 
     def lock_account(self):
         if self.device:
-            # TODO: Force Disconnect Devices?
-            pass
+            pass  # TODO: Force Disconnect Devices?
```

Comment:
hmmm, interesting TODO here. Might be worth raising an error or something for the time being, perhaps?

Reply:
I am a bit nervous to introduce any new crashing behavior here since this is part of the confirm activity loop.  See #1128 .